### ISSUE TITLE
torch 2.7.1 makes all npu unit tests failed

### ISSUE THREAD

**Issue by LeoLau94** [BD-Seed-HHW/xpu_graph#396]
### Describe the bug🐛

带上[这个acl graph内存复用单测](https://github.com/BD-Seed-HHW/xpu_graph/blob/c2a6b5be90c87da1e5ebc0b1e6b43c4f22ca28e8/tests/npu/test_acl_graph_mempool_reuse.py)跑，会导致后续的单测都失败
![Image](https://github.com/user-attachments/assets/5dc91a10-ece8-4a1f-ac09-fd3e04df044d)

不带上的话，都没问题
![Image](https://github.com/user-attachments/assets/865bce67-4b08-4c0d-802b-aa310a9fe5b5)

### Environment details

PyTorch: 2.7.1
Torch Npu: 2.7.1
XpuGraph: master

**Comment by wangqian108** [comment]
当前定位到是https://gitee.com/ascend/torchair/commit/9818eff91d926398e6bc2a733d044efe21629477这笔提交引起的用例失败。
把用例里的torch.inference_mode() 替换到 torch.no_grad() 用例可以执行成功。
修改了一处aclgraph初始化的逻辑后，当前可以跑通9个用例，仍然有4个失败，正在定位。

<img width="2236" height="245" alt="Image" src="https://github.com/user-attachments/assets/47748011-1fde-4ef7-9853-03832868409c" />

**Comment by LeoLau94** [comment]
> 当前定位到是https://gitee.com/ascend/torchair/commit/9818eff91d926398e6bc2a733d044efe21629477 这笔提交引起的用例失败。 把用例里的torch.inference_mode() 替换到 torch.no_grad() 用例可以执行成功。 修改了一处aclgraph初始化的逻辑后，当前可以跑通9个用例，仍然有4个失败，正在定位。
> 
> <img alt="Image" width="2000" height="245" src="https://private-user-images.githubusercontent.com/216635000/490863532-47748011-1fde-4ef7-9853-03832868409c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgxNzg3NTgsIm5iZiI6MTc1ODE3ODQ1OCwicGF0aCI6Ii8yMTY2MzUwMDAvNDkwODYzNTMyLTQ3NzQ4MDExLTFmZGUtNGVmNy05ODUzLTAzODMyODY4NDA5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxOFQwNjU0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04M2U1OWY2NDdmZmM3ODkzYWE1N2FmOTAxZTY2YTExZDg2YjcxYzc1YzgwZDU2NDYzZTdiOGI2NjliMTE3NjJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RE6yCrr58EAm9O2Ib_UgD6Jn_GbLmZBHixdaM5VO_wY">

请问一下，torch.inference_mode下，仍然会失败对吗？

**Comment by wangqian108** [comment]
> > 当前定位到是https://gitee.com/ascend/torchair/commit/9818eff91d926398e6bc2a733d044efe21629477 这笔提交引起的用例失败。 把用例里的torch.inference_mode() 替换到 torch.no_grad() 用例可以执行成功。 修改了一处aclgraph初始化的逻辑后，当前可以跑通9个用例，仍然有4个失败，正在定位。
> > <img alt="Image" width="2000" height="245" src="https://private-user-images.githubusercontent.com/216635000/490863532-47748011-1fde-4ef7-9853-03832868409c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgxNzg3NTgsIm5iZiI6MTc1ODE3ODQ1OCwicGF0aCI6Ii8yMTY2MzUwMDAvNDkwODYzNTMyLTQ3NzQ4MDExLTFmZGUtNGVmNy05ODUzLTAzODMyODY4NDA5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxOFQwNjU0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04M2U1OWY2NDdmZmM3ODkzYWE1N2FmOTAxZTY2YTExZDg2YjcxYzc1YzgwZDU2NDYzZTdiOGI2NjliMTE3NjJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RE6yCrr58EAm9O2Ib_UgD6Jn_GbLmZBHixdaM5VO_wY">
> 
> 请问一下，torch.inference_mode下，仍然会失败对吗？

是的，仍有4个用例失败。
另外测试到在aclgraph内存复用的用例下增加teardown_method，清dynamo缓存后所有用例都能成功。
class TestMemPoolReuse:
    ......

    def teardown_method(self):
        torch._dynamo.reset()

**Comment by wangqian108** [comment]
> > > > 当前定位到是https://gitee.com/ascend/torchair/commit/9818eff91d926398e6bc2a733d044efe21629477 这笔提交引起的用例失败。 把用例里的torch.inference_mode() 替换到 torch.no_grad() 用例可以执行成功。 修改了一处aclgraph初始化的逻辑后，当前可以跑通9个用例，仍然有4个失败，正在定位。
> > > > <img alt="Image" width="2000" height="245" src="https://private-user-images.githubusercontent.com/216635000/490863532-47748011-1fde-4ef7-9853-03832868409c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgxNzg3NTgsIm5iZiI6MTc1ODE3ODQ1OCwicGF0aCI6Ii8yMTY2MzUwMDAvNDkwODYzNTMyLTQ3NzQ4MDExLTFmZGUtNGVmNy05ODUzLTAzODMyODY4NDA5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwOTE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDkxOFQwNjU0MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04M2U1OWY2NDdmZmM3ODkzYWE1N2FmOTAxZTY2YTExZDg2YjcxYzc1YzgwZDU2NDYzZTdiOGI2NjliMTE3NjJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RE6yCrr58EAm9O2Ib_UgD6Jn_GbLmZBHixdaM5VO_wY">
> > > 
> > > 
> > > 请问一下，torch.inference_mode下，仍然会失败对吗？
> > 
> > 
> > 是的，仍有4个用例失败。 另外测试到在aclgraph内存复用的用例下增加teardown_method，清dynamo缓存后所有用例都能成功。 class TestMemPoolReuse: ......
> > ```
> > def teardown_method(self):
> >     torch._dynamo.reset()
> > ```
> 
> 额，我的意思可能没表达清楚，我想问的是，inference_mode不换成no_grads的话，那些在no_grads下能pass的测例仍然会失败么？

是的，就是上面那4个。

**Comment by LeoLau94** [comment]
@wangqian108 这个问题定位还有后续进展吗？

**Comment by wbigat** [comment]
目前确认与torchair的代码无关，是torch升级高版本之后才会出现，cudagraph基础接口也可以复现，后续在pytorch社区进一步讨论。

https://github.com/pytorch/pytorch/issues/166792
